cloud/wfm/src/result/BooleanResult.ts

```diff
@@ -0,0 +1,11 @@
+import {Result} from './index';
+
+/**
+ * Represents a {@link Result} that can be represented via a boolean value
+ * i.e. pass/fail, approval/rejection, etc.
+ */
+export default class BooleanResult implements Result {
```

Comment:
I want to spike results better in [926](https://issues.jboss.org/browse/RAINCATCH-926), so could we merge these as-is for now?

We can also validate how we can integrate a new paradigm for results when doing [928](https://issues.jboss.org/browse/RAINCATCH-927) and [929](https://issues.jboss.org/browse/RAINCATCH-929)